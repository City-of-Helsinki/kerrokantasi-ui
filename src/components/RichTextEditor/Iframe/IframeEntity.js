import React from 'react';
import PropTypes from 'prop-types';

const IframeEntity = (props) => {
  const {title, src} = props.contentState.getEntity(props.entityKey).getData();
  return (
    <div style={{overflow: "hidden"}}>
      <iframe title={title} src={src} />
    </div>
  );
};

IframeEntity.propTypes = {
  contentState: PropTypes.object,
  entityKey: PropTypes.string
};

export default IframeEntity;
