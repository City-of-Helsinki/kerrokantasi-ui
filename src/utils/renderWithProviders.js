/* eslint react/prop-types: 0 */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { LoginProvider } from 'hds-react';
import { IntlProvider } from 'react-intl';

import createStore from "../createStore";
import { userOidcConfig, apiTokenClientConfig } from './oidcConfig';

const loginProviderConfig = {
  userManagerSettings: userOidcConfig,
  apiTokensClientSettings: apiTokenClientConfig,
  sessionPollerSettings: { pollIntervalInMs: 10000 },
}

const renderWithProviders = (
  ui,
  { preloadedState = {}, locale = 'fi', history, store = createStore(preloadedState), ...renderOptions } = {},
) => {
  const Wrapper = ({ children }) => (
    <LoginProvider {...loginProviderConfig}>
      <IntlProvider locale={locale}>
        <Provider store={store}>{children}</Provider>
      </IntlProvider>
    </LoginProvider>
  );
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};


export default renderWithProviders;
