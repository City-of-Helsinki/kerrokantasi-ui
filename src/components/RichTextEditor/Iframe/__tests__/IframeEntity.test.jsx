import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import IframeEntity from '../IframeEntity';
import renderWithProviders from '../../../../utils/renderWithProviders';

const renderComponent = (propOverrides) => {
  const entity = {
    data: {
      title: 'some title',
      src: 'https://google.fi',
    },
    entityKey: 'IFRAME',
    getData() {
      return this.data;
    },
  };

  const contentState = { getEntity: () => entity };

  const props = {
    contentState,
    entityKey: 'IFRAME',
    ...propOverrides,
  };

  return renderWithProviders(
    <MemoryRouter>
      <IframeEntity {...props} />
    </MemoryRouter>,
  );
};

describe('<IframeEntity />', () => {
  it('should render correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
