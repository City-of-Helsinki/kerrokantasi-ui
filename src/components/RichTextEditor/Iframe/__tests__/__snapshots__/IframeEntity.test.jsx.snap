// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IframeEntity /> should render correctly 1`] = `
<div>
  <div
    style="overflow: hidden;"
  >
    <iframe
      src="https://google.fi"
      title="some title"
    />
  </div>
</div>
`;
